prov/verbs/src/fi_verbs.c

```diff
@@ -2123,6 +2134,50 @@ static int fi_ibv_copy_addr(void *dst_addr, size_t *dst_addrlen, void *src_addr)
 	return 0;
 }
 
+static int fi_ibv_msg_ep_setname(fid_t ep_fid, void *addr, size_t addrlen)
+{
+	struct fi_ibv_msg_ep *ep;
+	struct fi_info info;
+	int ret;
+
+	ep = container_of(ep_fid, struct fi_ibv_msg_ep, ep_fid);
+
+	if (addrlen != ep->src_addrlen) {
+		FI_INFO(&fi_ibv_prov, FI_LOG_EP_CTRL,"addrlen expected: %d, got: %d.\n",
+				ep->src_addrlen, addrlen);
+		return -FI_EINVAL;
+	}
+
+	info.src_addr = calloc(1, addrlen);
+	if (!info.src_addr)
+		return -FI_ENOMEM;
+
+	memcpy(info.src_addr, addr, addrlen);
+	info.src_addrlen = addrlen;
+
+
+	if (ep->dst_addr) {
+		info.dest_addrlen = ep->dst_addrlen;
+		info.dest_addr = calloc(1, info.dest_addrlen);
+		if (!info.dest_addr) {
+			ret = -FI_ENOMEM;
+			goto err;
+		}
+
+		memcpy(info.dest_addr, ep->dst_addr, info.dest_addrlen);
+	}
+
+	if (ep->id)
+		rdma_destroy_ep(ep->id);
```

Comment:
I would call fi_ibv_create_ep() before destroying the original id.  This will make it easier to undo things of create_ep fails.